v1/src/test/java/com/google/cloud/teleport/templates/BulkCompressorIT.java

```diff
@@ -72,7 +71,7 @@ public void testCompressGzip() throws IOException {
 
     Result result =
         new DataflowOperator(getDataflowClient())
-            .waitForCondition(
+            .waitForConditionAndFinish(
```

Comment:
I think I tried `AndFinish` before and it may cause some problems for batch jobs depending on timing (i.e., it is already completed and we call cancel). Usually batch jobs will finish by themselves (either completed or failed after 5 retries), so just waiting for condition may be enough.

Reply:
Ah, that makes sense! Have been writing only streaming template tests recently, didn't think about this.

Reply:
Changed.